bindings/rust/wasmedge-sdk/src/import.rs

```diff
@@ -326,6 +325,172 @@ impl ImportObjectBuilder {
 
         Ok(ImportObject(sys::ImportObject::WasmEdgeProcess(inner)))
     }
+
+    pub fn build_as_wasi_nn(self) -> WasmEdgeResult<ImportObject> {
+        let mut inner = sys::WasiNnModule::create()?;
+
+        // add func
+        for (name, func) in self.funcs.into_iter() {
+            inner.add_func(name, func);
+        }
+
+        // add global
+        for (name, global) in self.globals.into_iter() {
+            inner.add_global(name, global);
+        }
+
+        // add memory
+        for (name, memory) in self.memories.into_iter() {
+            inner.add_memory(name, memory);
+        }
+
+        // add table
+        for (name, table) in self.tables.into_iter() {
+            inner.add_table(name, table);
+        }
+
+        Ok(ImportObject(sys::ImportObject::Nn(inner)))
+    }
+
+    pub fn build_as_wasi_crypto_common(self) -> WasmEdgeResult<ImportObject> {
+        let mut inner = sys::WasiCryptoCommonModule::create()?;
+
+        // add func
+        for (name, func) in self.funcs.into_iter() {
+            inner.add_func(name, func);
+        }
+
+        // add global
+        for (name, global) in self.globals.into_iter() {
+            inner.add_global(name, global);
+        }
+
+        // add memory
+        for (name, memory) in self.memories.into_iter() {
+            inner.add_memory(name, memory);
+        }
+
+        // add table
+        for (name, table) in self.tables.into_iter() {
+            inner.add_table(name, table);
+        }
```

Comment:
I found each registry uses the same adding flow. Is it possible to reduce the duplicated codes?

Reply:
Yes. We can define a macro to do all of these things, but for now, this code snippet is not stabilized, we can refactor or change the design in the future. Therefore, it would be better for us to keep it there until it's stable. Thanks!

Reply:
Sure.